task/control-plane-agent/src/main.rs

```diff
@@ -290,8 +298,10 @@ impl NetHandler {
             self.tx_buf,
         );
 
-        meta.size = n as u32;
-        self.packet_to_send = Some(meta);
+        if n > 0 {
```

Comment:
The correctness here isn't obvious to me.  Should `handle_message` return an `Option<usize>` instead, to indicate whether or not to send a reply?

Reply:
Agreed, but `handle_message` is defined in another crate; I'll fix this in a followup PR.

Reply:
This is fixed by #927.